File: curry.py
from inspect import signature


def curry(function):
    arguments = list()
    number_of_args = len(signature(function).parameters)

    def wrapper(*args):
        arguments.extend(args)
        if len(arguments) == number_of_args:
            return function(*arguments)
        else:
            return wrapper

    return wrapper


def sum(a, b):
    return a + b


csum = curry(sum)
print(csum()(1)()(1))

File: curry.js
/**
 * @param {Function} fn
 * @return {Function}
 */
var curry = function (fn) {
    var inputs = []
    return function curried(...args) {
        inputs.push(...args)
        if (inputs.length === fn.length) {
            return fn(...inputs)
        } else {
            return curried
        }
    };
};


function sum(a, b) { return a + b; }
const csum = curry(sum);
console.log(csum()()(1, 3))
